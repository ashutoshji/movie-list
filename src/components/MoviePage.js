import React from 'react';
import MovieCard from './MovieCard';
import SearchBar from './SearchBar';
import Movies from './Movies'
import {URL,MOVIENOTFOUND} from '../data/Const.js';

class MoviePage extends React.Component{

state = {
        moviesList: [],
        searchInput: 'Dark'
    };

    componentDidMount=()=>{
       this.getPopularMovie();
    }

    getPopularMovie = () => {
        fetch(`${URL}&query=${this.state.searchInput}`).then(res => res.json())
				.then(res => {
				    this.setState({
			            moviesList: res.results
			        });
					console.log('Success:', JSON.stringify(res)) 
				}).catch(error => console.error('Error:', error));
    }

     searchForMovie = event => {
            event.preventDefault();
            if(event){
               this.setState({
	            searchInput: event.target.value
		        });
	        if(event.length>2){
	        	  this.getPopularMovie();
	        	}
            }
    };


	render(){
	   const { moviesList ,searchInput} = this.state;
	   return(
        <div className="container">
		  <SearchBar
		    searchInput={searchInput}
		    handleChange={(event) => this.searchForMovie(event)}/>
        	{moviesList.length > 0 ? (
        	          <Movies movies={moviesList}/>
                ) : (
                    <p>
                        {MOVIENOTFOUND}
                    </p>
                )}
        </div>
	   )
	}
}

export default MoviePage;
