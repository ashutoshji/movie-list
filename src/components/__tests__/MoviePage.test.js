
// Auto-generated do not edit


/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import MoviePage from '../MoviePage';


describe('MoviePage test', () => {
  it('MoviePage should match snapshot', () => {
    const component = renderer.create(<MoviePage
       />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
