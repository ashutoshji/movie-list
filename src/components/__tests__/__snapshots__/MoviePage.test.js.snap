// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviePage test MoviePage should match snapshot 1`] = `
<div
  className="container"
>
  <div
    className="search__container"
  >
    <p
      className="search__title"
    >
      Go ahead, Search any movie
    </p>
    <input
      className="search__input"
      onChange={[Function]}
      placeholder="Search for a movie"
      type="text"
      value="Dark"
    />
  </div>
  <p>
    Couldn't find any movie. Please search again using another search criteria.
  </p>
</div>
`;
